showmore_dyn/showmore_dyn.js

```diff
@@ -0,0 +1,81 @@
+$(document).ready(function(){
+        handleNewWallItemBodies();
+
+        var mutationObserver = new MutationObserver(function(mutations) {
+                handleNewWallItemBodies();
+        });
+        mutationObserver.observe($("#content")[0], { attributes: false, characterData: false, childList: true, subtree: true, attributeOldValue: false, characterDataOldValue: false });
+});
+
+function handleNewWallItemBodies() {
+        $('.wall-item-body:not(.showmore-done)').each(function(i, el) {
+                $(el).addClass('showmore-done');
+                if ($(el).has('button.content-filter-button').length > 0) {
+                        $(el).removeClass('limitable');
+                        return;
+                }
+
+                var itemId = $(el).attr('id');
+                addHeightToggleHandler(itemId);
+                var limited = processHeightLimit(itemId);
+
+                if (!limited) {
+                        var mutationObserver = new MutationObserver(function(mutations) {
+                                var limited = processHeightLimit(itemId);
+                                if (limited) {
+                                        mutationObserver.disconnect()
+                                }
+                        });
+                        mutationObserver.observe(el, { attributes: true, characterData: true, childList: true, subtree: true, attributeOldValue: true, characterDataOldValue: true });
+
+                        $(el).imagesLoaded().then(function(){
+                                processHeightLimit(itemId);
+                        });
+                }
+        });
+}
+
+function addHeightToggleHandler(id) {
+        var itemIdSel = "#" + id;
+        var itemId = parseInt(id.replace("wall-item-body-", ""));
+        $(itemIdSel).data("item-id", itemId);
+        var wrapperId = "wall-item-body-wrapper-" + itemId;
+        var wrapperIdSel = "#" + wrapperId;
+        var toggleId = "wall-item-body-toggle-" + itemId;
+        var toggleIdSel = "#" + toggleId;
+
+        $(itemIdSel).wrap('<div id="' + wrapperId + '" class="wall-item-body-wrapper"></div>');
+        $(wrapperIdSel).append('<div class="wall-item-body-toggle" data-item-id="' + itemId + '" id="' + toggleId + '" ><a href="javascript:void(0)" class="wall-item-body-toggle-text">Show more ...</a></div>');
```

Comment:
For general accessibility, `<a href="javascript:void(0)">...</a>` should be replaced by a `<button type="button">...</button>`. It's a little more challenging for styles because of the default button styles, but I trust you to find the right CSS directives to keep the same look.